library/core/src/fmt/mod.rs

```diff
@@ -430,6 +438,14 @@ impl<'a> Arguments<'a> {
             _ => None,
         }
     }
+
+    /// Same as [`Arguments::as_str`], but will only return `Some(s)` if it can be determined at compile time.
+    #[must_use]
+    #[inline]
+    fn as_const_str(&self) -> Option<&'static str> {
```

Comment:
"const_str" is a very confusing name since `const` has a meaning in Rust and it's not this. We have very deliberately called the intrinsic `is_val_statically_known`, not `is_val_const` or anything like that. Would be nice if the functions using that intrinsic would follow the same pattern. :)

Reply:
So... `as_statically_known_str`?

Reply:
Yes, that would be better IMO. (I'm also open to suggestions for a shorter term that captures this, but "constant" is too misleading IMO. In Rust, whether something is a "constant" is a language guarantee, and we have things like `const fn` that relate to that. In contrast, "statically known" is a very unstable notion depending on tons of factors that impact how good the compiler is able to analyze some piece of code.)